svgelements/svgelements.py

```diff
@@ -2943,6 +2949,47 @@ def _calc_lengths(self, error=ERROR, min_depth=MIN_DEPTH, segments=None):
         else:
             self._lengths = [each / self._length for each in lengths]
 
+    def npoint(self, positions, error=ERROR):
+        """
+        Find a points between 0 and 1 within the shape. Numpy acceleration allows points to be an array of floats.
+        """
+        if not _NUMPY:
+            return [self.point(pos) for pos in positions]
+
+        segments = self.segments(False)
+        if len(segments) == 0:
+            return None
+        # Shortcuts
+        if self._length is None:
+            self._calc_lengths(error=error, segments=segments)
+        xy = np.empty((len(positions), 2), dtype=float)
+        if self._length == 0:
+            i = int(round(positions * (len(segments) - 1)))
+            point = segments[i].point(0.0)
+            xy[:] = point
+            return xy
+
+        # Find which segment the point we search for is located on:
+        segment_start = 0
+        for index, segment in enumerate(segments):
+            segment_end = segment_start + self._lengths[index]
+            position_subset = ((segment_start <= positions) & (positions < segment_end))
+            v0 = positions[position_subset]
+            if not len(v0):
+                continue  # Nothing matched.
```

Comment:
This turned out to be a bug.

This requires `segment_start = segment_end` or the non-matched segment doesn't count. So by the end we're missing a few segments of length if those segments were small enough to not be hit by the call.

Reply:
Well spotted 👍🏻